### PR TITLE
Use FileObject for downloads

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke usage of `FileObject` in downloads. I'm having some concerns about this approach and think I'm starting to conflate `django-soil`, `FileObject`, and the currrent behavior. I think it may be best to talk offline about a good approach to this.

buddy: @orangejenny 

corresponding PR: https://github.com/dimagi/dimagi-utils/pull/294

### CODE REVIEW

**Comment by benrudolph** [comment]
Thought about this a bit more:

`soil`/`DownloadBase` - Handles async downloads with progress management. It also handles the rendering of the download as a response.
`FileObject`/`CachedObject` - Provides an interface for storing/retrieving objects temporarily. No support for rendering to a response, that is strictly the duty of the program using the object

The part that was bugging me was that I was using `soil` to render the HTTP response of a `FileObject`. So I was only using one small part of `soil`, yet rendering an entire instance of `DownloadBase`. Part of me feels like the logic for rendering a file as a response should be abstracted into its own module. There are instances (such as this one) where the file already exists. In this case as an attachment in Couch. It'll be nice to have utilities to:
1. Temporarily store binary data that is too large, or not logical to store in redis
2. Work with binary data to return a proper HttpResponse, whether that be `TransferHttpResponse` or `StreamingHttpResponse` and handle setting the response headers given a file stream.

I think it may be out of scope to do that in these PRs, but at least something worth thinking about.

**Comment by snopoke** [comment]
That sounds like a good long term approach.

**Comment by benrudolph** [comment]
Ok, this is potentially g2g. will throw it up on staging and dig in a bit more when i get back to boston. i also miscalculated our slowest queries on redis. they're in seconds, not minutes now (not sure if i had misread it before as well). i still think this is an improvement as our hit to miss ratio will improve without large binary blobs in redis. and it's probably best to handle the downloading of the files via django-transfer.

File: corehq/apps/app_manager/views.py [hunk 1]
```diff
@@ -77,7 +77,8 @@
 from corehq.apps.translations import system_text_sources
 from corehq.apps.translations.models import Translation
 from corehq.util.view_utils import set_file_download, absolute_reverse, json_error
-from dimagi.utils.django.cached_object import CachedObject
+from corehq.util.view_utils import set_file_download
```
Comment (dimagimon) [inline corehq/apps/app_manager/views.py]: F811 redefinition of unused 'set_file_download' from line 79

**Pull request closed without merging**